packages/trpc/server/createContext.ts

```diff
@@ -12,6 +13,7 @@ import type { CreateNextContextOptions } from "@trpc/server/adapters/next";
 type CreateContextOptions = CreateNextContextOptions | GetServerSidePropsContext;
 
 export type CreateInnerContextOptions = {
+  sourceIp?: string;
```

Comment:
Maybe @zomars @keithwillcode can figure out a better way to have this not be optional 

It needs to be optional due to the fact CreateSsgHelpers needs the full context - however we have no way of getting the IP into that. they also use the infered types so we can't omit in one place and not the other.